c10/util/order_preserving_flat_hash_map.h

```diff
@@ -685,11 +682,22 @@ class sherwood_v3_table : private EntryAlloc, private Hasher, private Equal
     convertible_to_iterator erase(const_iterator to_erase)
     {
         EntryPointer current = to_erase.current;
+        remove_from_list(current);
         current->destroy_value();
         --num_elements;
+
         for (EntryPointer next = current + ptrdiff_t(1); !next->is_at_desired_position(); ++current, ++next)
         {
+            // if an entry is being removed, and there are other entries with the
+            // same hash, the other entries get moved to their desired position by
+            // reinserting.
             current->emplace(next->distance_from_desired - 1, std::move(next->value));
+            // the value of "current" is now "next", so remove it from the list,
```

Comment:
The word `it` here is ambiguous in both instances. Also this english and the code below say the same thing. It would be better if the comment explains why this needed rather than what it is doing.

Reply:
I think this logic is pretty unclear. Isn't the action being done to replace `current` in the list with `next`? If so having a primitive`replace(Node* to_be_replaced, Node* with_me)`would be more clear, even if the implementation just did the same exact calls.

Reply:
> the value of "current" is now "next 

i thought said something different then the code 🤷